extensions/markdown-language-features/src/languageFeatures/copyFiles/shared.ts

```diff
@@ -221,6 +221,7 @@ export function appendToLinkSnippet(
 export function createUriListSnippet(
 	document: SkinnyTextDocument,
 	uris: readonly vscode.Uri[],
+	uriStrings?: readonly string[],
```

Comment:
Instead of adding a new argument, maybe take an array of objects with both values: `{ uri: Uri, text?: string }

Reply:
That was my initial thought, however I would have to edit the insertResource file a it and I don't want to mess with that code especially during endgame.